src/test/java/spoon/test/issue4357/TestIssue4357.java

```diff
@@ -0,0 +1,27 @@
+package spoon.test.issue4357;
+
+import org.junit.jupiter.api.Test;
+import spoon.Launcher;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.net.URLClassLoader;
+import java.nio.file.Paths;
+
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+
+public class TestIssue4357 {
+    
+    @Test
+    public void testClasspathURLWithSpaces() throws MalformedURLException {
+        // contract: launcher can handle spaces in classpath URL
+        Launcher launcher = new Launcher();
+        URL[] classpath = {
+                Paths.get("src/test/resources/path with spaces").toUri().toURL()
+        };
+        ClassLoader classLoader = new URLClassLoader(classpath);
+        launcher.getEnvironment().setInputClassLoader(classLoader);
+        launcher.getEnvironment().setNoClasspath(false);
+        assertDoesNotThrow(() -> launcher.buildModel());     
```

Comment:
a stronger assert would be checking for the presence of CtType Foo in the model. WDYT?

Reply:
OK, that's next on my list.

Reply:
Ah, this was a bit more complicated than expected because sppon doesn't load java files via classloaders. So I added a jar file that is used during the test. After some trial and error (I accidentally compiled the jar using JDK 17 and other oversights) the test finally passed in all tested configurations.